### PR TITLE
Use @percy/cli 

### PR DESCRIPTION
**Pull request by Robdel12**
## What is this?

There is a new kid on the block, [`@percy/cli`](https://github.com/percy/cli), that takes what `@percy/agent` has done and makes it more robust, more modular, more extensible, and more feature-rich than `@percy/agent` can ever be in its current state. This PR brings `@percy/cli` to this SDK by refactoring the entire SDK from the ground up, including new tests, new CI config, and other niceties. The new `@percy/cli` has many benefits over `@percy/agent`, so I won't list all of them here. But importantly, there are a few necessary breaking changes.

### Breaking Changes

- `@percy/agent` is no longer a dependency, _obviously_, but neither is `@percy/cli`. This enables users to install the SDK as a development dependency without requiring the entire SDK umbrella on install (including a lengthy chromium download). When using `npx`, users will have to use `npx @percy/cli ...` rather than `npx percy`, as the latter refers to`@percy/agent` and has gaps in its version history that would prevent some future releases.

- To enable normal package script behavior with `percy <command>`, users will have to install the CLI alongside the SDK – `npm install @percy/cli @percy/puppeteer`. The new CLI does not download Chromium on install, and instead does so the first time asset discovery is started. So even when choosing to install the CLI as a development dependency, it has a much smaller footprint than it's predecessor.

- The format of Percy configuration files has changed. Common options can be viewed in the [`@percy/cli-command`](https://github.com/percy/cli/tree/master/packages/cli-command#percy-configuration) package, and previous configuration options can be migrated to the new format using [`percy config:migrate`](https://github.com/percy/cli/tree/master/packages/cli-config#percy-configmigrate-filepath-output).

### Repo Changes:

- Tests were re-written to be functional. The previous tests were visual only and in reality were testing the behavior of `@percy/agent` using the SDK rather than testing the SDK itself. Coverage reporting has also been added and is a requirement for tests to pass to have complete coverage.

- GitHub Actions are now the default CI and other CIs do not need to be tested within since this repo should be testing the SDK and not CI integrations. Separate workflows exist for linting and tests so their failures can be reported separately (rather than having tests be cancelled due to failed linting).

- The changelog and release processes have also been updated to fully utilize GitHub features such as tags and releases. The semantic release bot and several semantic release packages were replaced with [release-drafter](https://github.com/release-drafter/release-drafter) which utilizes PRs and labels to create changelogs found alongside each release on the releases page.

- Automatic package releasing is now a semi-manual process due to bots not always being able to correctly guess the proper version number and sometimes releasing the incorrect version. To release, the version number should be bumped appropriately and then the automatically created draft release can be published from the releases page. In the future, we can automate the version bump to reflect the release version to make this process smoother, while still requiring a human to approve the release.

### CODE REVIEW

**Comment by Robdel12** [comment]
- [ ] Update README

**Review by wwilsman — APPROVED** [review]
🎮

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: v1/${{ runner.os }}/node-12/${{ hashFiles('**/yarn.lock') }}
+          restore-keys: v1/${{ runner.os }}/node-12/
+      - run: yarn
+      - run: yarn build
+      - run: npm publish
```
Comment (wwilsman) [inline .github/workflows/release.yml:24]: We should change this to include `--tags next` so we don't release this as latest

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+version-resolver:
+  major:
+    labels:
+      - breaking
+  minor:
+    labels:
+      - feature
+      - enhancement
+  default: patch
+template: '$CHANGES'
```
Comment (wwilsman) [inline .github/release-drafter.yml:30]: Similar to the npm tag, we can add `prerelease: true` here to mark draft releases automatically as prereleases

**Pull request merged**